synapse/axon.py

```diff
@@ -913,14 +1050,81 @@ async def _populate(self, sha256, genr, size=None):
             return size
 
     async def _saveFileGenr(self, sha256, genr):
+
         size = 0
+
         for i, byts in enumerate(genr):
+
             size += len(byts)
-            lkey = sha256 + i.to_bytes(8, 'big')
-            self.blobslab.put(lkey, byts, db=self.blobs)
+
+            ikey = i.to_bytes(8, 'big')
+            okey = size.to_bytes(8, 'big')
+
+            self.blobslab.put(sha256 + ikey, byts, db=self.blobs)
+            self.blobslab.put(sha256 + okey, ikey, db=self.offsets)
+
             await asyncio.sleep(0)
+
         return size
 
+    def _offsToIndx(self, sha256, offs):
+        lkey = sha256 + offs.to_bytes(8, 'big')
+        for offskey, indxbyts in self.blobslab.scanByRange(lkey, db=self.offsets):
+            return int.from_bytes(offskey[32:], 'big'), indxbyts
+
+    async def getByteRange(self, sha256, soff, eoff):
```

Comment:
Why is this not exposed over telepath?

Reply:
Because we don't have a use case for it yet and i wasn't sure it was the calling convention we wanted facing the outside :)